dev/io.openliberty.microprofile.telemetry.2.0.logging.internal/src/io/openliberty/microprofile/telemetry20/logging/internal/OpenTelemetryLogHandler.java

```diff
@@ -0,0 +1,279 @@
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.logs.LogRecordBuilder;
+import io.opentelemetry.api.logs.Severity;
+import io.opentelemetry.semconv.SemanticAttributes;
+
+@Component(name = OpenTelemetryLogHandler.COMPONENT_NAME, service = {
+	Handler.class, ServerQuiesceListener.class }, configurationPolicy = ConfigurationPolicy.OPTIONAL, property = { "service.vendor=IBM" })
+public class OpenTelemetryLogHandler extends Collector implements ServerQuiesceListener {
+
+	private static final TraceComponent tc = Tr.register(OpenTelemetryLogHandler.class);
+
+	public static final String COMPONENT_NAME = "io.openliberty.microprofile.telemetry20.logging.internal.OpenTelemetryLogHandler";
+	
+	private TelemetryLogEmitter tle = null;
+	
+	private OpenTelemetry openTelemetry;
+	
+	@Override
+	@Reference(name = EXECUTOR_SERVICE, service = ExecutorService.class)
+	protected void setExecutorService(ServiceReference<ExecutorService> executorService) {
+	    executorServiceRef.setReference(executorService);
+	}
+
+	@Override
+	protected void unsetExecutorService(ServiceReference<ExecutorService> executorService) {
+	    executorServiceRef.unsetReference(executorService);
+	}
+
+	@Override
+	@Activate
+	protected void activate(ComponentContext cc, Map<String, Object> configuration) {
+	    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+	    	    Tr.debug(tc, "In activate()");
+	    }
+	    this.openTelemetry = OpenTelemetryAccessor.getOpenTelemetryInfo("io.openliberty.microprofile.telemetry.runtime").getOpenTelemetry();
+	    // Configure message as the only source.
+	    Map<String, Object> config = setSourceListToConfig(configuration);
+	    super.activate(cc, config);
+	}	
+	
+	@Override
+	@Deactivate
+	protected void deactivate(ComponentContext cc, int reason) {
+	    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+	    	    Tr.debug(tc, "In deactivate()");
+	    }
+	    super.deactivate(cc, reason);
+	}
+
+	@Override
+	@Modified
+	protected void modified(Map<String, Object> configuration) {
+	    if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+	            Tr.debug(tc, "In modified()");
+	    }
+	    // Configure message as the only source.
+	    Map<String, Object> config = setSourceListToConfig(configuration);
+	    super.modified(config);
+
+	}
+    
+	@Override
+	public String getHandlerName() {
+	    return COMPONENT_NAME;
+	}
+
+	@Override
+	public Object formatEvent(String source, String location, Object event, String[] tags, int maxFieldLength) {
+	    LogRecordBuilder builder = null;
+	    String eventType = CollectorJsonUtils.getEventType(source, location);
+	    if (eventType.equals(CollectorConstants.MESSAGES_LOG_EVENT_TYPE)) {
+	        LogTraceData logData = (LogTraceData) event;
+	        if (logData.getMessage().contains("scopeInfo:")) {
+	            // Log a warning - This should only be the case when the logs exporter is set to logging.
```

Comment:
Comment says to log a warning but no warning is logged.

Reply:
Yes, this is a To-Do for GA. For Beta, it is not needed.